data_model/src/isi.rs

```diff
@@ -22,7 +22,10 @@ use crate::{seal, Level, Registered};
 /// Instructions allows to change the state of `Iroha`.
 /// All possible instructions are implementors of this trait, excluding
 /// [`InstructionBox`] which is just a wrapper.
-pub trait Instruction: Into<InstructionBox> + seal::Sealed {}
+pub trait Instruction: Into<InstructionBox> + seal::Sealed {
+    /// [`Encode`] [`Self`] as [`InstructionBox`]
+    fn encode_as_instruction_box(&self) -> Vec<u8>;
```

Comment:
For future code readers I'd add some explanation why do we need this hack

Reply:
> What about to use [EncodeLike](https://docs.rs/parity-scale-codec/latest/parity_scale_codec/trait.EncodeLike.html)?

I see, you probably don't want for `ConcreteIsi::encode()` to be equal to `ConcreteIsi::encode_as_instruction_box()`, right? But why not actually...?

Reply:
> I see, you probably don't want for `ConcreteIsi::encode()` to be equal to `ConcreteIsi::encode_as_instruction_box()`, right? But why not actually...?

yes, I don't think that's a good idea. And I can't just implement `EncodeLike`, I have to first implement custom `Encode`/`Decode` for every concrete ISI

Reply:
> And I can't just implement EncodeLike, I have to first implement custom Encode/Decode for every concrete ISI

Yes, but a custom proc macro could simpify that

Reply:
actually, I'm becoming fond of this idea. I'm not sure if there is any downside except that we will always add one byte to serialized form of concrete ISI. But then again I think that we're always serializing `InstructionBox`, never concrete instruction. This will bring us a lot of benefits

Reply:
I chose a hybrid approach. This could be a foot gun if we ever introduce individual instructions in the API considering that it wouldn't match the schema we provide SDKs for encoding